cmd/root.go

```diff
@@ -108,13 +108,11 @@ This allows users to schedule kubernetes workloads on nodes that aren't running
 			rootContextCancel()
 		}()
 
-		c1, c2, err := setupHTTPServer(rootContext, apiConfig)
+		cancelHTTP, err := setupHTTPServer(rootContext, apiConfig)
 		if err != nil {
 			log.G(rootContext).Fatal(err)
```

Comment:
Call `cancelHTTP`?

Reply:
I think we should just have cancel be nil when there is an error and call cancel in the setup function. Updating this.